File: app-fx/src/main/java/chessgame/app/fx/util/FXMLHelper.java
package chessgame.app.fx.util;

import chessgame.app.fx.Controller;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

import java.net.URL;

public final class FXMLHelper {

    private FXMLHelper() {
        super();
    }

    public static Scene createScene(Stage stage, String viewTitle, String view, String... styles) throws Exception {
        URL v = getResource(view);
        URL[] s = getResources(styles);

        return createScene(stage, viewTitle, v, s);
    }

    private static Scene createScene(Stage stage, String viewTitle, URL view, URL... styles) throws Exception {
        Parent root = createView(stage, viewTitle, view, styles);

        return new Scene(root);
    }

    public static <V extends Parent> V createView(Stage stage, String viewTitle, String view, String... styles) throws Exception {
        URL v = getResource(view);
        URL[] s = getResources(styles);

        return createView(stage, viewTitle, v, s);
    }

    private static <V extends Parent> V createView(Stage stage, String viewTitle, URL view, URL... styles) throws Exception {
        FXMLLoader fxml;
        Controller controller;
        V root;

        fxml = new FXMLLoader(view);
        root = fxml.load();
        controller = fxml.getController();

        for (URL style : styles) {
            if (style != null)
                root.getStylesheets().add(style.toExternalForm());
        }

        stage.setTitle(viewTitle);

        controller.setStage(stage);

        return root;
    }

    private static URL getResource(String r) {
        return FXMLHelper.class.getResource(r);
    }

    private static URL[] getResources(String... r) {
        if (r == null) {
            r = new String[]{};
        }

        URL[] rs = new URL[r.length];

        for (int i = 0; i < rs.length; i++) {
            rs[i] = FXMLHelper.class.getResource(r[i]);
        }

        return rs;
    }
}

File: app-kotlin-fx/src/main/kotlin/chessgame/app/fx/util/FXMLHelper.kt
package chessgame.app.fx.util

import chessgame.app.fx.api.Controller
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage
import java.net.URL


class FXMLHelper {

    companion object {
        @Throws(Exception::class)
        fun createScene(stage: Stage?, viewTitle: String, view: String, vararg styles: String) =
                createScene(stage, viewTitle, getResource(view), getResources(*styles))

        @Throws(Exception::class)
        private fun createScene(stage: Stage?, viewTitle: String, view: URL, styles: Array<URL?>) =
                Scene(createView(stage, viewTitle, view, styles))

        @Throws(Exception::class)
        private fun <V : Parent> createView(stage: Stage?, viewTitle: String, view: URL, styles: Array<URL?>): V {
            val fxml = FXMLLoader(view)
            val root: V = fxml.load()
            val controller: Controller? = fxml.getController<Controller>()

            styles.forEach { style -> root.stylesheets.add(style?.toExternalForm()) }

            stage?.title = viewTitle

            controller?.stage = stage

            return root
        }

        private fun getResource(r: String): URL = FXMLHelper::class.java.getResource(r)

        private fun getResources(vararg r: String): Array<URL?> = r.map(FXMLHelper::class.java::getResource).toTypedArray()
    }
}